### ISSUE TITLE
The next best guess does not match the real sample point.

### ISSUE THREAD

**Issue by linghuix** [bayesian-optimization/BayesianOptimization#528]
**Describe the bug**

I noticed the next best guess from utility function does not match the real one.

**To Reproduce**

I used the same code as shown in the document https://bayesian-optimization.github.io/BayesianOptimization/1.5.1/visualization.html

**Expected behavior**

The next best guess from ucb utility function should match the real sample point. The code doesn't explore around x=-0.2 . If there is a high peak at that point, the code will miss it.

**Screenshots**
![image](https://github.com/user-attachments/assets/a43fee47-b5fc-471d-badf-2b398d0b5b5a)
![image](https://github.com/user-attachments/assets/a9c183e7-47db-4b5a-991f-f34faebb82f6)
![image](https://github.com/user-attachments/assets/120f2186-ffa6-4d77-9361-5ecce0c4a29f)
![image](https://github.com/user-attachments/assets/344fdad3-3e00-46b7-b909-195ccd96e579)

**Environment (please complete the following information):**
 - OS: Windows 11
 - `python` Version 3.9.20
 - `numpy` Version 2.0.2
 - `scipy` Version 1.13.1
 - `bayesian-optimization` Version 1.5.1

**Comment by linghuix** [comment]
I forget to attach this figure. It is obvious that the next best guess has highest value but optimizer doesn't choose it. There could be something wrong here.
![image](https://github.com/user-attachments/assets/3e79cb43-61f0-4002-888d-258c883b0a73)

I compared my results with the document and found the different happens from 4th step. Code is exactly the same, quite wired.
![image](https://github.com/user-attachments/assets/67e6078e-e92c-4746-8233-80d3dfa35a74)

**Comment by till-m** [comment]
Hi @linghuix,

If I run the notebook I don't observe this phenomenon. Are you sure you're running exactly that notebook, with nothing changed?

NB: Sometimes the GP fitting process fails (usually due to little data), which can then cause this can happen. However, I don't think that's the case here.

**Comment by linghuix** [comment]
Thanks for your reply. 

After I carefully compared my code with that notebook, I noticed the only difference is that I used `optimizer.maximize(init_points=0, n_iter=1)` instead of `
optimizer.maximize(init_points=0, n_iter=1, acquisition_function=acq_function)`.

I am curious that what the default acq function is if I donot pass one. Also, in the Visulazation Document, the following code does not specify the acq function. Is it a typo?
![image](https://github.com/user-attachments/assets/4fa558ae-0337-4f56-8b9f-16a72b1bd967)
